### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by tyronjung-quizlet**
### Ticket
Link to ticket: **No linked ticket**

### What has been done
- Rename some files.
- Update docs with examples.
- Update docs with new features.

### CODE REVIEW

File: MAIN_API.md [hunk 1]
```diff
@@ -0,0 +1,398 @@
+# Overview
+
+The key distinction made by Hammock is between _method containers_ and _functions_. Method containers comprise classes and objects. Functions comprise methods and global functions. Mocking a method container allows the developer to mock the behavior of any method within it. Hence, the mocking interface for method containers is called a _method mock container_. When a method is mocked through a method mock container, it instantiates a _function mock_, which is what actually mocks the method's behavior. The function mock is the unit of mocking within Hammock, since Hammock ultimately mocks functions.
```
Comment (lexidor) [inline MAIN_API.md:3]: Method containers comprise _of_ classes and objects?

File: MAIN_API.md [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+	- `@param T $object`
+	- `@param string $methodName`
+	- `@return FunctionMock`
+	- <details><summary>Example</summary><p>
+
+		```hack
+		$alice = new User('Alice');
+		$bob = new User('Bob');
+
+		using getNameSpy = Hammock\spy_object_method($alice, 'getName');
```
Comment (lexidor) [inline MAIN_API.md]: Missing `$` for `$getNameSpy`.

File: PERSISTENT_API.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
+# Overview
+
+The persistent API is very similar to the main API, but the returned interfaces are not disposable. This means that the `using` keyword is omitted, and the resulting mocks are not automatically cleaned up at the end of the block scope. The persistent API is especially useful for repeatedly creating the same mocks during test setup. Because these mocks will not be cleaned up at the end of the setup, the mocked behavior may be shared across many different tests. However, this means that the persistent mocks have to be cleaned up manually, preferably during teardown. Otherwise, there could be a memory leak.
```
Comment (lexidor) [inline PERSISTENT_API.md]: I'd say: `there could be a memory leak and the mocked behavior is retained until the end of the request / CLI invocation.`

**Review by lexidor — APPROVED** [review]
I did not read the persistent api examples, since I reckoned they are carbon copies of the non-persitents, but without the `using`.

File: MAIN_API.md (hunk 1 above)
Comment (lexidor) [reply MAIN_API.md:3]: Idem for `Function comprise methods and global functions`.

Comment (tyronjung-quizlet) [reply MAIN_API.md:3]: In my mind it's "are composed of" or "comprise" (based on dictionary.com).

Comment (lexidor) [reply MAIN_API.md:3]: Lexidor English isn't always the best.

Comment (tyronjung-quizlet) [reply MAIN_API.md:3]: TBH `comprise` always gets me too.

**Pull request merged**